packages/atomic/src/components/search/atomic-search-box/atomic-search-box.tsx

```diff
@@ -516,7 +517,15 @@ export class AtomicSearchBox {
 
   private getAndFilterLeftSuggestionElements() {
     const suggestionElements = this.getSuggestionElements(this.leftSuggestions);
-    return getUniqueItemsByProperties(suggestionElements, ['query']);
+    // We want to remove duplicates elements for the same `query` property, but only when it's not empty.
+    // Since the custom search box suggestion system does not enforce the query property (it's optional),
+    // it's a system that custom search box suggestions component will use to provide a "title" or separator section.
+    // We don't want to detect multiple separator section with no query as duplicate
+    return suggestionElements.filter(
+      (suggestionElement, i, self) =>
+        isNullOrUndefined(suggestionElement.query) ||
+        i === self.findIndex((other) => other.query === suggestionElement.query)
+    );
```

Comment:
nipic, but your function is O(n^2): you're going thru the array twice.
But you could go thru it only once.
I think also that this implementation, albeit more verbose is also clearer and would avoid the comment.


```suggestion
    return suggestionElements.filter((suggestionElement) => {
      if (isNullOrUndefined(suggestionElement.query)) {
        return true;
      }
      if (queryAlreadyFound.has(suggestionElement.query)) {
        return false;
      } else {
        queryAlreadyFound.add(suggestionElement.query);
        return true;
      }
    });
```